src/lib/stepper/stepper.spec.ts

```diff
@@ -0,0 +1,559 @@
+    });
+
+    it('should go to previous available step when the previous button is clicked', () => {
+      checkPreviousStepperButton(stepperComponent, fixture);
+    });
+
+    it('should set the correct step position for animation', () => {
+      checkStepPosition(stepperComponent, fixture);
+    });
+
+    it('should support keyboard events to move and select focus', () => {
+      let stepHeader = fixture.debugElement.queryAll(By.css('.mat-vertical-stepper-header'));
+      checkKeyboardEvent(stepperComponent, fixture, stepHeader);
+    });
+  });
+
+  describe('linear vertical stepper', () => {
+    let fixture: ComponentFixture<LinearMdVerticalStepperApp>;
+    let testComponent: LinearMdVerticalStepperApp;
+    let stepperComponent: MdVerticalStepper;
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(LinearMdVerticalStepperApp);
+      fixture.detectChanges();
+
+      testComponent = fixture.componentInstance;
+      stepperComponent = fixture.debugElement
+          .query(By.css('md-vertical-stepper')).componentInstance;
+    });
+
+    it('should have true linear attribute', () => {
+      expect(stepperComponent.linear).toBe(true);
+    });
+
+    it('should not move to next step if current step is not valid', () => {
+      expect(testComponent.oneGroup.get('oneCtrl')!.value).toBe('');
+      expect(testComponent.oneGroup.valid).toBe(false);
+      expect(stepperComponent.selectedIndex).toBe(0);
+
+      let stepHeaderEl = fixture.debugElement
+          .queryAll(By.css('.mat-vertical-stepper-header'))[1].nativeElement;
+
+      checkLinearStepperValidity(stepHeaderEl, stepperComponent, testComponent, fixture);
+    });
+  });
+});
+
+function checkSelectionChangeOnHeaderClick(stepperComponent:
+                                               MdHorizontalStepper | MdVerticalStepper,
+                                           fixture: ComponentFixture<any>,
+                                           stepHeader: DebugElement[]) {
+  expect(stepperComponent.selectedIndex).toBe(0);
+
+  // select the second step
+  let stepHeaderEl = stepHeader[1].nativeElement;
+  stepHeaderEl.click();
+  fixture.detectChanges();
+
+  expect(stepperComponent.selectedIndex).toBe(1);
+
+  // select the third step
+  stepHeaderEl = stepHeader[2].nativeElement;
+  stepHeaderEl.click();
+  fixture.detectChanges();
+
+  expect(stepperComponent.selectedIndex).toBe(2);
+}
+
+function checkExpandedContent(stepperComponent: MdHorizontalStepper | MdVerticalStepper,
+                              fixture: ComponentFixture<any>,
+                              stepContent: DebugElement[]) {
+  let stepContentEl = stepContent[0].nativeElement;
+  expect(stepContentEl.getAttribute('aria-expanded')).toBe('true');
+
+  stepperComponent.selectedIndex = 1;
+  fixture.detectChanges();
+
+  expect(stepContentEl.getAttribute('aria-expanded')).toBe('false');
+  stepContentEl = stepContent[1].nativeElement;
```

Comment:
Nit: It's a bit hard to follow when you're naming two different step elements with the same name. Maybe create a new `let` for this - `secondStepEl`?